daemon/oci_linux.go

```diff
@@ -264,6 +264,12 @@ func setCapabilities(s *specs.Spec, c *container.Container) error {
 	s.Process.Capabilities.Bounding = caplist
 	s.Process.Capabilities.Permitted = caplist
 	s.Process.Capabilities.Inheritable = caplist
+	// setUser has already been executed here
+	// if non root drop capabilities in the way execve does
+	if s.Process.User.UID != 0 {
+		s.Process.Capabilities.Effective = []string{}
```

Comment:
So when the user specifies `--cap-(add|drop)` on the CLI, this is available in the bounding set?  I thought that we also wanted the effective set to remain consistent with what was requested from the end-user across the "_user context_" switch.

Reply:
Ah nevermind, I had to read more carefully `CAPABILITIES(7)`. Yes, this seems like the behavior we should have had from the start.